packages/cst-dts-gen/src/generate.ts

```diff
@@ -0,0 +1,95 @@
+import { map, upperFirst, flatten } from "@chevrotain/utils"
+import {
+  CstNodeTypeDefinition,
+  PropertyTypeDefinition,
+  PropertyArrayType
+} from "./model"
+
+export type GenDtsOptions = {
+  includeTypes: boolean
+  includeVisitorInterface: boolean
+  visitorInterfaceName: string
+}
+
+export function genDts(
+  model: CstNodeTypeDefinition[],
+  options: GenDtsOptions
+): string {
+  let contentParts: string[] = []
+
+  if (options.includeTypes || options.includeVisitorInterface) {
+    contentParts = contentParts.concat(
+      `import type { CstNode, ICstVisitor, IToken } from "chevrotain";`
+    )
+  }
+
+  if (options.includeTypes) {
+    contentParts = contentParts.concat(
+      flatten(map(model, (node) => genCstNodeTypes(node)))
+    )
+  }
+
+  if (options.includeVisitorInterface) {
+    contentParts = contentParts.concat(
+      genVisitor(options.visitorInterfaceName, model)
+    )
+  }
+
+  return contentParts.join("\n\n")
```

Comment:
should we use `os.EOL` instead of hard-coded "\n"?
- https://nodejs.org/api/os.html#os_os_eol

But I think that would mean this code won't work in the browser...
Should we conditionally `require` os.EOL` and fallback to `\n` if that does not exist?

But maybe I am over-complicating things? as this API is not really meant to be used in the browser?
Maybe we can provide an **optional** parameter in `GenDtsOptions` for the newline and let the consumers
provide whatever they want, and fallback to "\n" if none was provided.

Reply:
I'm thinking that's over-complicating things.  And would be better handled by the user running a formatter over the output (i.e. prettier)